server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -239,6 +254,15 @@ long buildTookInMillis() {
 
     @Override
     protected void doExecute(Task task, SearchRequest searchRequest, ActionListener<SearchResponse> listener) {
+        // only if task is of type CancellableTask and support cancellation on timeout, treat this request eligible for timeout based
+        // cancellation. There may be other top level requests like AsyncSearch which is using SearchRequest internally and has it's own
+        // cancellation mechanism. For such cases, the SearchRequest when created can override the createTask and provide the necessary
+        // flag to indicate it and bypass this mechanism
```

Comment:
We don't want to add support for `AsyncSearch` since it has its own cancellation mechanism. Refer: [1](https://github.com/opensearch-project/asynchronous-search/blob/main/src/main/java/org/opensearch/search/asynchronous/rest/RestDeleteAsynchronousSearchAction.java) and [2](https://github.com/opensearch-project/asynchronous-search/blob/bd31e80adf6d52c1b4662d0d2cc9b30d8ae14309/src/main/java/org/opensearch/search/asynchronous/service/AsynchronousSearchService.java#L350). Given these are long running requests, a timeout based cancellation will not make sense instead probably an explicit event would be more useful, hence bypassing those.